editor/editor_vcs_api.h

```diff
@@ -0,0 +1,25 @@
+#ifndef EDITOR_VCS_API_H
+#define EDITOR_VCS_API_H
+
+#include "core/ustring.h"
+#include "core/object.h"
+
+class EditorVCSAPI {
```

Comment:
I would rename this as EditorVCS. There is no need to mention API here, as abstract classes are basically APIs.